File: client/src/app/core/models/IngredientRecipe.ts
import { Ingredient } from "./Ingredient";
import { UnitMeasure } from "../enums/UnitMeasure";

export class IngredientRecipe{

    constructor(
        public measure:UnitMeasure, 
        public quantity: number, 
        public ingredientPrice: number, 
        public totalCost: number, 
        public ingredients: Ingredient

    ){}
}

File: RecipesApp_/RecipesApp.Core/Entities/IngredientRecipe.cs
using RecipesApp.Common.Enums;

namespace RecipesApp.Database.Entities
{
    public class IngredientRecipe
    {
        public Recipe Recipes { get; set; }
        public int RecipesId { get; set; }

        public Ingredient Ingredients { get; set; }
        public int IngredientsId { get; set; }

        public int Quantity { get; set; }
        public UnitMeasureType Measure { get; set; }

    }
}
